app/models/comment.rb

```diff
@@ -12,6 +12,8 @@ class Comment < ApplicationRecord
   TITLE_DELETED = "[deleted]".freeze
   TITLE_HIDDEN = "[hidden by post author]".freeze
 
+  URI_REGEXP = %r{(?<scheme>https?://)?(?<host>.+?)(?<port>:\d+)?$}.freeze
```

Comment:
> Looks like this matches a bit too eagerly. When giving it a URL inside a line of surrounding text (as one might expect in a post or comment), it matches the entire line

In our case, this will be a non issue because shorten_urls! takes in processed_html, so the text above will look like this -
``` html
this is my new thing hosted at <a href="https://example.com/omg?wat=lol">https://example.com/omg?wat=lol</a>. what do you think?
```
Since the method only takes in content inside the anchor tag, only the url will be passed through regex scan.

That being said, this regex may cause unexpected bugs in the future. URI.regexp breaks it down a lot, so we won't be able to use that in our case. I'll play around with URI_REGEXP and make it more bug proof.

Reply:
@squarebat Oh interesting. It might work, then. I was wondering how it was passing the test you wrote with the broad matching but `shorten_urls` accounting for that makes sense. As long as this won't cause a regression I'm good with it. 🙂